staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go

```diff
@@ -97,8 +104,29 @@ type CSINodeInfo struct {
 	CSIDrivers []CSIDriverInfo `json:"csiDrivers" patchStrategy:"merge" patchMergeKey:"driver"`
 }
 
-// CSIDriverInfo contains information about one CSI driver installed on a node.
+// CSIDriverInfo holds information about one CSI driver installed on a node.
 type CSIDriverInfo struct {
+	// spec is the specification of CSIDriverInfo
+	Spec CSIDriverInfoSpec `json:"spec"`
+
+	// status is the current status of CSIDriverInfo
+	Status CSIDriverInfoStatus `json:"status"`
+}
+
+// CSIDriverInfoStatus holds information about the status of one CSI driver installed on a node
+type CSIDriverInfoStatus struct {
+	// available is a boolean representing whether the driver has been installed
+	// on this node or not.
+	Available bool `json:"available"`
+
+	// requiredDriverPluginMechanism is set by Kubelet to indicate to
+	// external storage controllers the required mechanism to use for any in-tree
+	// plugins replaced by this driver
+	RequiredDriverPluginMechanism string `json:"requiredDriverPluginMechanism"`
```

Comment:
I think "required" and "driver" are not adding much clarity, I'd omit them and clarify what kind of plugin it's talking about: `VolumePluginMechanism`.